### DISCUSSION TITLE
[V4] How to override theme styles?

### DISCUSSION THREAD

**Discussion by LawJolla** [tailwindlabs/tailwindcss#13165] [Help] [answered]
Say that I have the following:

```
// globals.css
@import "tailwindcss";

@theme {
  ...
  --color-mono-200: var(--color-slate-200);
}

:root {
  ...
  --color-mono-200: var(--color-stone-200);
}
```

Color mono 200 (e.g. bg-mono-200) will not update. 

```
:root {
  ...
  --color-slate-200: var(--color-stone-200);
}
```

does work, but obviously that kind of defeats the purpose.  Thanks!

**Comment by adamwathan** [comment]
Hey! This works as I'd expect:

<img width="1338" alt="image" src="https://github.com/tailwindlabs/tailwindcss/assets/4323180/17a1442d-bf9e-4b28-8212-f43d004d9b83">

Can you share more details or better yet an example project?

**Reply by adamwathan** [reply, marked as answer]
My mistake I read your example wrong — so we inline the values of the variables into your utilities by default to avoid unnecessary variable lookups in the browser, and because out of the box it makes more sense to me to think of them as constants, not as variables.

If you want your theme values to point to actual run time variables, you need to define them as variables and update the underlying variable, like in my example above.

**Reply by wongjn** [reply]
Could be related to #13114?

I think they may be after something like:

```css
@import "tailwindcss";

@theme {
  --color-mono-200: var(--mono-200);
}

:root {
  --mono-200: var(--color-slate-200);
}

[data-theme="stone"] {
  --mono-200: var(--color-stone-200);
}
```

But for its syntax to be more like:

```css
@import "tailwindcss";

@theme {
  --color-mono-200: var(--color-slate-200);
}

[data-theme="stone"] {
  --color-mono-200: var(--color-stone-200);
}
```

The default theme configuration is surfaced in a `:root` rule, but custom configuration is not, which leads to a double declaration. For example compare a core color and a custom color:

```css
.bg-slate-200 {
  background-color: var(--color-slate-200);
}
```
vs
```css
.bg-mono-200 {
  background-color: var(--color-slate-200);
}
```
Should it not be perhaps:
```css
.bg-mono-200 {
  background-color: var(--color-mono-200);
}
```
Where
```css
:root {
  --color-mono-200: var(--color-slate-200);
}
```
is automatically generated from the `@theme`?

**Reply by adamwathan** [reply]
Yeah will reply to the other thread eventually when I have time to reply in detail — we made the decision to inline the values instead of making everything variables because it's always possible to point them at variables if you like but if we make them variables by default there's no way for you to opt out.

I did a poll on Twitter the other day to see what people thought and it was close but ~60% preferred having it inline vs. point to a variable:

https://twitter.com/adamwathan/status/1765496800289415621

**Reply by LawJolla** [reply]
Thanks @adamwathan !

It sounds like the best approach is to initialize it like the following...?

```
@theme {
  ...
    --color-mono-200: var(--color-setMono-200);
}

:root {
  --color-setMono-200: var(--color-slate-200);
 }

```

Update:  Just read the discussion above... really helped @wongjn  and Adam, thank you!

**Reply by wongjn** [reply]
Hmm, I see that point for things like spacing that I would not imagine changing with theme switching, but I feel like its a bit more of a gray area with colors. To me, it would feel more consistent to having core values and custom values both emit CSS variables, but you're smarter than me so I will cede to your judgement :)
